src/server/src/main/java/io/cassandrareaper/resources/RepairRunResource.java

```diff
@@ -663,17 +663,15 @@ public Response listRepairRuns(
             : context.storage.getClusters();
 
       List<RepairRunStatus> runStatuses = Lists.newArrayList();
-      for (final Cluster clstr : clusters) {
-        Collection<RepairRun> runs = context.storage.getRepairRunsForCluster(clstr.getName(), limit);
-
-        runStatuses.addAll(
-            (List<RepairRunStatus>) getRunStatuses(runs, desiredStates)
-                .stream()
-                .filter((run) -> !keyspace.isPresent()
-                    || ((RepairRunStatus)run).getKeyspaceName().equals(keyspace.get()))
-                .collect(Collectors.toList()));
-      }
-
+      List<RepairRun> repairRuns = Lists.newArrayList();
+      clusters.forEach(clstr -> repairRuns.addAll(context.storage.getRepairRunsForCluster(clstr.getName(), limit)));
```

Comment:
I think we should call `context.storage.getRepairRunsForClusterPrioritise()` running here, otherwise the limits will be applied as before and any long running repair could get hidden by more recent finished runs.

Reply:
Ah, I think that's why my manual tests were failing! Agreed, I'll fix this. Thanks for the suggestion.